lib/shale/schema/json_generator/base.rb

```diff
@@ -12,15 +12,18 @@ class Base
         # @api private
         attr_reader :name
 
-        # Return nullable
+        attr_reader :mapping
+
+        # Set nullable
         #
         # @api private
         attr_writer :nullable
 
-        def initialize(name, default: nil)
+        def initialize(name, default: nil, mapping: nil)
```

Comment:
Can we accept here, just schema, I this we don't need full mapping object.

Reply:
I actually think we may be able to use schema directly everywhere instead of passing mapping. I think I originally did this because I saw we were passing `mapping.name` and I was thinking that I could just pass in `mapping` and get everything we needed from it, but that didn't work out.